### PR TITLE
mac-robber: init at 1.02

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Description of changes
mac-robber: a digital investigation tool that collects data from allocated files in a mounted file system.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/mac-robber/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # add GCC hardening.
+    (fetchpatch {
+      url = "https://salsa.debian.org/pkg-security-team/mac-robber/-/raw/b6a59d78e2f58fbfab7f1b3ed9b72531d28693ca/debian/patches/10_add-GCC-hardening.patch";
+      hash = "sha256-OyZbNha82iuaFOnyWyXTD38UakQa8WKlWL3tO9z9/l8=";
+    })
+  ];
+
+  buildInputs = [
+    glibc.static
+  ];
```
Comment (emilytrau) [inline pkgs/by-name/ma/mac-robber/package.nix:27]: glibc is not required, just need to ignore the default `-static` flag
```suggestion
  makeFlags = [
    "CC=cc"
    "GCC_OPT="
  ];
```

File: pkgs/by-name/ma/mac-robber/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp mac-robber $out/bin/
```
Comment (emilytrau) [inline pkgs/by-name/ma/mac-robber/package.nix:32]:
```suggestion
    install -D mac-robber $out/bin/mac-robber
```

**Comment by emilytrau** [comment]
The makefile has a seperate `linux` target that sets `-D_FILE_OFFSET_BITS=64`. Not sure if it affects functionality.

File: pkgs/by-name/ma/mac-robber/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A digital investigation tool that collects data from allocated files in a mounted file system";
+    mainProgram = "mac-robber";
+    homepage = "https://www.sleuthkit.org/mac-robber/";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
+    license = licenses.gpl2Plus;
```
Comment (emilytrau) [inline pkgs/by-name/ma/mac-robber/package.nix:42]:
```suggestion
    license = licenses.gpl2Only;
```

**Comment by D3vil0p3r** [comment]
> The makefile has a seperate `linux` target that sets `-D_FILE_OFFSET_BITS=64`. Not sure if it affects functionality.

I tested it to a folder and it seems to work with no errors.

**Comment by emilytrau** [comment]
Result of `nixpkgs-review pr 289532` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mac-robber</li>
  </ul>
</details>

**Review by emilytrau — APPROVED** [review]

**Pull request merged**